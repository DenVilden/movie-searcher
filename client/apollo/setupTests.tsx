import { render } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components';
import { theme } from '../pages/_app';

// eslint-disable-next-line jest/prefer-spy-on
HTMLElement.prototype.scrollIntoView = jest.fn();

type RenderApolloOptions = {
  mocks?: MockedResponse[];
  addTypeName?: boolean;
};

export const renderApollo = (
  element: React.ReactElement,
  { mocks = [], addTypeName = false }: RenderApolloOptions = {},
) =>
  render(
    <MockedProvider addTypename={addTypeName} mocks={mocks}>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>,
  );

export * from '@testing-library/react';
