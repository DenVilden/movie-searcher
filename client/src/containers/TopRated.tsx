import { useRef } from "react";
import styled from "styled-components";
import ErrorMessage from "./ErrorMessage";
import MoviesBox from "../components/MoviesBox";
import { useGetTopRatedLazyQuery } from "../generated/queries.generated";
import Pagination from "../components/Pagination";
import { TopRated } from "../generated/types";

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

type Props = {
  initialData: TopRated;
};

const TopRatedContainer = ({ initialData }: Props) => {
  const [refetch, { data, error }] = useGetTopRatedLazyQuery();

  const element = useRef<HTMLDivElement>(null!);

  if (error) return <ErrorMessage error={error} />;

  return (
    <Wrapper ref={element}>
      <MoviesBox
        movies={data?.topRated.results || initialData.results}
        title="Top Rated"
      />
      <Pagination
        currentPage={data?.topRated.page || initialData.page}
        refetch={refetch}
        scrollToTop={element}
        totalPages={data?.topRated.total_pages || initialData.total_pages}
      />
    </Wrapper>
  );
};

export default TopRatedContainer;
