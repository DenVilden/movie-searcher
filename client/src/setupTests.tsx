import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components';
import { theme } from './pages/_app';
import defaultResolvers from './graphql/resolvers';

window.HTMLElement.prototype.scrollIntoView = jest.fn();

type RenderApolloOptions = {
  mocks?: MockedResponse[];
  addTypeName?: boolean;
  cache?: any;
  resolvers?: any;
};

export const renderApollo = (
  element: React.ReactElement,
  { mocks, addTypeName, resolvers, cache }: RenderApolloOptions = {}
) =>
  render(
    <MockedProvider
      addTypename={addTypeName}
      cache={cache}
      mocks={mocks}
      resolvers={resolvers || defaultResolvers}
    >
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>
  );

export * from '@testing-library/react';
