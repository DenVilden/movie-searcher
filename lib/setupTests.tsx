/* eslint-disable import/no-extraneous-dependencies */
import { render } from '@testing-library/preact'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { ThemeProvider, createMuiTheme } from '@material-ui/core/styles'
import '@testing-library/jest-dom/extend-expect'

const theme = createMuiTheme()

type RenderApolloOptions = {
  addTypeName?: boolean
  mocks?: MockedResponse[]
}

export function renderApollo(
  element: React.ReactElement,
  { mocks = [], addTypeName = false }: RenderApolloOptions = {},
) {
  return render(
    <MockedProvider addTypename={addTypeName} mocks={mocks}>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>,
  )
}

export { screen, fireEvent } from '@testing-library/preact'
