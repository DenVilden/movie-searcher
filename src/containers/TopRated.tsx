import React, { useRef } from 'react';
import styled from 'styled-components';
import { LinearProgress } from '@material-ui/core';
import ErrorMessage from '../components/ErrorMessage';
import MoviesBox from '../components/MoviesBox/MoviesBox';
import { useGetTopRatedLazyQuery } from '../__generated__';
import Pagination from '../components/Pagination/Pagination';
import { TopRated } from '../types/types';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

type Props = {
  initialData: TopRated;
};

const TopRatedContainer = ({ initialData }: Props) => {
  const [refetch, { data, loading, error }] = useGetTopRatedLazyQuery();

  const element = useRef<HTMLDivElement>(null!);

  if (loading) return <LinearProgress color="secondary" />;

  if (error) return <ErrorMessage>{error.message}</ErrorMessage>;

  return (
    <Wrapper ref={element}>
      <MoviesBox
        movies={data?.topRated.results || initialData.results}
        title="Top Rated"
      />
      <Pagination
        currentPage={data?.topRated.page || initialData.page}
        refetch={refetch}
        scrollToTop={element}
        totalPages={data?.topRated.total_pages || initialData.total_pages}
      />
    </Wrapper>
  );
};

export default TopRatedContainer;
