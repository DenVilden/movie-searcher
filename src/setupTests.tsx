import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MockedProvider, MockedResponse } from '@apollo/react-testing';
import { ThemeProvider } from 'styled-components';
import { createMuiTheme } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import defaultResolvers from './graphql/resolvers';

window.HTMLElement.prototype.scrollIntoView = jest.fn();

const theme = createMuiTheme();

type RenderApolloOptions = {
  mocks?: MockedResponse[];
  addTypeName?: boolean;
  cache?: any;
  resolvers?: any;
};

export const renderApollo = (
  element: React.ReactElement,
  { mocks, addTypeName, resolvers, cache }: RenderApolloOptions = {}
) =>
  render(
    <MockedProvider
      addTypename={addTypeName}
      cache={cache}
      mocks={mocks}
      resolvers={resolvers || defaultResolvers}
    >
      <ThemeProvider theme={theme}>
        <Router>{element}</Router>
      </ThemeProvider>
    </MockedProvider>
  );

export * from '@testing-library/react';
