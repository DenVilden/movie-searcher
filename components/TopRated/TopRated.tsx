import styled from 'styled-components';
import { useRef } from 'react';
import { LinearProgress } from '@material-ui/core';
import { useGetTopRatedLazyQuery, GetTopRatedQuery } from '../../apollo';
import { ErrorMessage, Pagination, MoviesBox } from '..';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

interface Props {
  initialData: GetTopRatedQuery;
}

const TopRated = ({ initialData }: Props) => {
  const [fetchTopRated, { data, error, loading }] = useGetTopRatedLazyQuery({
    fetchPolicy: 'cache-and-network',
    nextFetchPolicy: 'cache-first',
  });
  const topRatedElement = useRef<HTMLDivElement | null>(null);

  if (error) return <ErrorMessage error={error.message} />;

  if (loading) return <LinearProgress color="secondary" />;

  return (
    <Wrapper ref={topRatedElement}>
      <MoviesBox
        movies={data?.topRated.results || initialData.topRated.results}
        title="TopRated"
      />
      <Pagination
        currentPage={data?.topRated.page || initialData.topRated.page}
        refetch={(page: number) => fetchTopRated({ variables: { page } })}
        element={topRatedElement}
        totalPages={
          data?.topRated.total_pages || initialData.topRated.total_pages
        }
      />
    </Wrapper>
  );
};

export default TopRated;
