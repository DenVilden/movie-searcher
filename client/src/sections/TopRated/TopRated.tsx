import { useRef } from "react";
import { MoviesBox, ErrorMessage, Pagination } from "../../components";
import { useGetTopRatedLazyQuery, TopRated } from "../../graphql";
import { Wrapper } from "./TopRated.styles";

type Props = {
  initialData: TopRated;
};

const TopRatedContainer = ({ initialData }: Props) => {
  const [refetch, { data, error }] = useGetTopRatedLazyQuery();

  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  const element = useRef<HTMLDivElement>(null!);

  if (error) return <ErrorMessage error={error} />;

  return (
    <Wrapper ref={element}>
      <MoviesBox
        movies={data?.topRated.results || initialData.results}
        title="Top Rated"
      />
      <Pagination
        currentPage={data?.topRated.page || initialData.page}
        refetch={refetch}
        scrollToTop={element}
        totalPages={data?.topRated.total_pages || initialData.total_pages}
      />
    </Wrapper>
  );
};

export default TopRatedContainer;
