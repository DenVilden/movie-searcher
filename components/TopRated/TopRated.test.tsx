import TopRated from './TopRated';
import { renderApollo } from '../../lib/setupTests';

const mocks = {
  topRated: {
    total_pages: 20,
    page: 1,
    results: [
      {
        id: 1,
        title: 'test',
        release_date: '2020',
        poster_path: null,
        vote_average: 10,
      },
    ],
  },
};

describe('topRated', () => {
  it('should take a snapshot', () => {
    const { asFragment } = renderApollo(<TopRated initialData={mocks} />);

    const element = asFragment();

    expect(element).toMatchSnapshot();
  });
});
