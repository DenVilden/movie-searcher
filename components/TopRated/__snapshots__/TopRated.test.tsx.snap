// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`topRated should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="TopRated__Wrapper-bvwfvl-0 fUksrc"
  >
    <div
      class="MuiPaper-root MoviesBox__Root-elk2wj-0 eoGsCK MuiPaper-elevation10 MuiPaper-rounded"
      padding="1"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        TopRated
      </h4>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MoviesBox__Wrapper-elk2wj-1 bmgJbm MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-6 MuiGrid-grid-md-3"
          data-testid="search-result"
        >
          <div
            class="MuiPaper-root MuiCard-root MovieCard__StyledCard-sc-5grlib-0 fQwzBi MuiPaper-elevation10 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              data-testid="card-button"
              tabindex="0"
              type="button"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                    style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                  />
                </div>
                <img
                  alt="test"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
              <div
                class="MuiCardContent-root MovieCard__StyledCardContent-sc-5grlib-1 bcACtj"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  test
                </h6>
                <p
                  class="MuiTypography-root MovieCard__StyledTypography-sc-5grlib-2 fIsZBD MuiTypography-body1 MuiTypography-colorTextSecondary"
                >
                  <span
                    class="MovieCard__IconWrapper-sc-5grlib-3 hWHmux"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                     10
                  </span>
                </p>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <nav
      aria-label="pagination navigation"
      class="MuiPagination-root Pagination__StyledPagination-ht8t9i-0 cBtSkt"
    >
      <ul
        class="MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-current="true"
            aria-label="page 1"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-selected"
            tabindex="0"
            type="button"
          >
            1
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 2"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
            tabindex="0"
            type="button"
          >
            2
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 3"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
            tabindex="0"
            type="button"
          >
            3
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 4"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
            tabindex="0"
            type="button"
          >
            4
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 5"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
            tabindex="0"
            type="button"
          >
            5
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <div
            class="MuiPaginationItem-root MuiPaginationItem-ellipsis"
          >
            â€¦
          </div>
        </li>
        <li>
          <button
            aria-label="Go to page 20"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
            tabindex="0"
            type="button"
          >
            20
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
